### ISSUE TITLE
Vectorized calls over multiple variables do not work

### ISSUE THREAD

**Issue by eguiraud** [eguiraud/correctionlib-gradients#43]
I think because `np.vectorize` chokes on JAX returning the N gradients in a tuple.

A repro:

```python
@pytest.mark.parametrize("jit", [False, True])
def test_complex_formula_vectorized(jit):
    cg = CorrectionWithGradient(schemas["complex-formula"])
    evaluate = jax.jit(cg.evaluate) if jit else cg.evaluate
    # pass in different kinds of arrays/collections
    y = jnp.array(2.)
    for x in [1.0, 2.0], np.array([1., 2.]), jnp.array([1., 2.]):
        values, grads = np.vectorize(jax.value_and_grad(evaluate, argnums=[0,1]))(x, y)
        assert len(values) == 2
        assert np.allclose(values, [9.963647609000805, 14.7853985])
        assert len(grads) == 88
        assert np.allclose(grads, [2.0, 4.0, 6.0])
```

**Comment by eguiraud** [comment]
Using `jax.vmap` instead of `np.vectorize` solves the problem.
